consensus-engine/src/overlay/membership.rs

```diff
@@ -0,0 +1,39 @@
+// std
+
+// crates
+use rand::prelude::SliceRandom;
+use rand::SeedableRng;
+use rand_chacha::ChaCha20Rng;
+
+// internal
+use crate::overlay::CommitteeMembership;
+use crate::NodeId;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct FisherYatesShuffle {
+    entropy: [u8; 32],
+}
+
+impl FisherYatesShuffle {
+    pub fn new(entropy: [u8; 32]) -> Self {
+        Self { entropy }
+    }
+
+    pub fn shuffle<T: Clone>(elements: &mut [T], entropy: [u8; 32]) {
```

Comment:
to be nitpicky, I don't think there's any guarantee or contract in the `shuffle` api about using fisher yates or any other specific algorithm (or having consistent results across versions).
If there's no other high quality library available I would suggest we note this in a comment here

Reply:
Yeah, let me clear that out.

Reply:
I found a crate with an implementation, but damn the API is quite bad. So I just implemented it myself 😅 (it was fairly easy though).